mne/beamformer/tests/test_dics.py

```diff
@@ -321,7 +328,11 @@ def test_apply_dics_timeseries():
     filters = make_dics(evoked.info, fwd_surf, csd20, label=label, reg=reg)
 
     # Sanity checks on the resulting STC after applying DICS on epochs.
-    stcs = apply_dics_epochs(epochs, filters)
+    # Also test whether stcs contain src
+    with warnings.catch_warnings(record=True) as wrn:
```

Comment:
I don't think you have yet checked to see if there ever *is* a warning, so I would do:
```
del filters['src_type']  # emulate 0.16 behavior to cause warning
with pytest.warns(RuntimeWarning, match='src should not be None'):
    ...
```

Reply:
Okay, after the changes with ``src_type`` etc., this now breaks before reaching this error. So to ensure backward compatibility, e.g., with saved ``filters`` or the like, I see it the following now:
- we should make ``src_type`` optional within ``_apply_lcmv``
- throw a warning in ``_apply_lcmv`` if ``src_type`` is not in ``filters``; deprecate?

How it is now, everyone that saved a ``filters`` object to disk will not be able to run a beamformer on it anymore.
WDYT, @larsoner and @agramfort ?

Reply:
I thought the plan was to have it `warnings.warn` for these older ones

Reply:
(and use the private function / old method for guessing)

Reply:
Yes - but: in this iteration now, the ``_make_stc`` call in the beamformer code requires ``src_type``. So if ``filters`` does not have it, it crashes. So I think we need to allow the function call without ``src_type`` still in ``apply_lcmv`` etc. - and then ``_make_stc`` subfunctions throw the warning about supplying ``stc``.

Reply:
Agreed

Reply:
I will change it accordingly then!